common/persistence/versionHistory.go

```diff
@@ -21,174 +21,513 @@
 package persistence
 
 import (
+	"bytes"
 	"fmt"
+
 	"github.com/uber/cadence/.gen/go/shared"
 	"github.com/uber/cadence/common"
-	"reflect"
 )
 
-// NewVersionHistory initializes new version history
-func NewVersionHistory(items []VersionHistoryItem) VersionHistory {
-	if len(items) == 0 {
-		panic("version history items cannot be empty")
+// NewVersionHistoryItem create a new version history item
+func NewVersionHistoryItem(
+	inputEventID int64,
+	inputVersion int64,
+) *VersionHistoryItem {
+
+	if inputEventID < 0 || inputVersion < 0 {
+		panic("invalid version history item event ID / version")
```

Comment:
can you also emit the values of inputEventID and inputVersion for debugging ?

Reply:
sure